File: src/main/java/com/ullgren/modern/simple3d/render/texture/NoTexture.java
package com.ullgren.modern.simple3d.render.texture;

/**
 * Null-object {@link Texture}: applies only Gouraud shading with no textural variation.
 */
public final class NoTexture implements Texture {

  @Override
  public int applyPacked(double wx, double wy, double wz,
                         int baseR, int baseG, int baseB,
                         float shade) {
    int r = Math.min(255, (int) (baseR * shade));
    int g = Math.min(255, (int) (baseG * shade));
    int b = Math.min(255, (int) (baseB * shade));
    return 0xFF000000 | (r << 16) | (g << 8) | b;
  }
}

File: src/web/src/render/texture/NoTexture.ts
import type { Texture } from './Texture.js';

/** Null-object texture: applies only Gouraud shading, no surface pattern. */
export class NoTexture implements Texture {
  prepare(_wx: number, _wy: number, _wz: number): number { return 0; }

  applyPacked(
    _texValue: number,
    baseR: number, baseG: number, baseB: number,
    shade: number,
  ): number {
    const r = Math.min(255, baseR * shade) | 0;
    const g = Math.min(255, baseG * shade) | 0;
    const b = Math.min(255, baseB * shade) | 0;
    return (r << 16) | (g << 8) | b;
  }
}
